ocaml/xapi/xapi_gpumon.ml

```diff
@@ -46,4 +46,22 @@ module Nvidia = struct
     let get = Gpumon_client.Client.Nvidia.get_pgpu_metadata in
     [key, get dbg pgpu_pci_address]
 
+  let get_pgpu_vm_compatibility ~__context ~vm ~pgpu =
+    let open Gpumon_client in
+    let dbg = Context.string_of_task __context in
+    let vm_domid = Int64.to_int (Db.VM.get_domid ~__context ~self:vm) in
+    let cur_pgpu_address =
+      Db.VM.get_VGPUs ~__context ~self:vm
+      (* currently we support only 1 nvidia vgpu *)
+      |> List.hd
+      |> (fun self -> Db.VGPU.get_resident_on ~__context ~self)
+      |> (fun self -> Db.PGPU.get_PCI ~__context ~self)
+      |> (fun self -> Db.PCI.get_pci_id ~__context ~self)
```

Comment:
I don't think so, unless the caller of this function has the relevant data in hand.
(If it were an SQL database then it could be done with a single query joining across the various tables.)